File: CodeForces/101-200/141/A/141A.py
# Problem 141A: Amusing Joke
# https://codeforces.com/contest/141/problem/A
# Executed with: Python 3.11.2
# Linux version: Linux kernel 6.1.0-10-amd64
# Submitted on: September 27th, 2024
# Codeforces language used: PyPy 3.10 (7.3.15, 64bit)

from collections import Counter

name1, name2, all_letters = input(), input(), input()

full_names: Counter[str] = Counter()
full_names.update([char for char in name1+name2])

hit = False

for char in all_letters:
    if char in full_names:
        full_names.subtract(char)
        full_names =+ full_names
    else:
        print("NO")
        hit = True
        break

if len(full_names) == 0 and not hit:
    print("YES")
elif not hit:
    print("NO")

File: CodeForces/101-200/141/A/141A.rs
// Problem 141A: Amusing Joke
// https://codeforces.com/contest/141/problem/A
// Compiled with: rustc 1.85.1 (4eb161250 2025-03-15) (built from a source tarball)
// Linux version: Linux kernel 6.17.0-5-generic
// Submitted on: October 26th, 2025
// Codeforces language used: Rust 1.89.0 (2024)

fn main() {
    let name1 = std::io::stdin().lines().next().unwrap().unwrap();
    let name2 = std::io::stdin().lines().next().unwrap().unwrap();
    let all_letters = std::io::stdin().lines().next().unwrap().unwrap();
    let all_names = name1 + &name2;

    let mut frequency: Vec<i32> = vec![0; 26];

    for c in all_letters.chars() {
        let index = ((c as i32)-65) as usize;
        frequency[index] += 1;
    }

    for c in all_names.chars() {
        let index = ((c as i32)-65) as usize;
        frequency[index] -= 1;
    }

    for val in &frequency {
        if *val != 0 {
            println!("NO");
            return;
        }
    }
    println!("YES");
}
